lib/internal/Magento/Framework/File/Mime.php

```diff
@@ -108,6 +108,9 @@ public function getMimeType($file)
 
         if (function_exists('mime_content_type')) {
             $result = $this->getNativeMimeType($file);
+        } else {
+            $imageInfo = getimagesize($file);
+            $result = $imageInfo['mime'];
         }
 
         if (null === $result && isset($this->mimeTypes[$extension])) {
```

Comment:
Also not clear why this part wasn't executed? 😕

Reply:
It was executing but returning `application/octet-stream` cause the tmp file doesn't have any extension so we can't check the mime type if fileinfo extension is disabled and without file extension and its leads to file validation exception
https://github.com/magento/magento2/blob/5cd4345b76f7e7806124195aa8647e5c365cce1f/lib/internal/Magento/Framework/File/Mime.php#L113-L117